### PR TITLE
[AMDGPU][Clang] Builtin for GLOBAL_LOAD_LDS on GFX940

### PR DESCRIPTION
**Pull request by shiltian**
Fixes: SWDEV-459212

### CODE REVIEW

File: llvm/include/llvm/IR/IntrinsicsAMDGPU.td [hunk 1]
```diff
@@ -2466,23 +2466,24 @@ def int_amdgcn_perm :
...
-  "", [SDNPMemOperand]>;
+class AMDGPUGlobalLoadLDS :
+  ClangBuiltin<"__builtin_amdgcn_global_load_lds">,
+  Intrinsic <
+    [],
+    [LLVMQualPointerType<1>,            // Base global pointer to load from
+     LLVMQualPointerType<3>,            // LDS base pointer to store to
+     llvm_i32_ty,                       // Data byte size: 1/2/4 (/12/16 for gfx950)
+     llvm_i32_ty,                       // imm offset (applied to both global and LDS address)
+     llvm_i32_ty],                      // auxiliary data (imm, cachepolicy (bit 0 = glc/sc0,
```
Comment (rampitec) [inline llvm/include/llvm/IR/IntrinsicsAMDGPU.td:2477]: Keep description of only sc0, sc1, and scc? It is not supported except on gfx940 anyway.

Comment (shiltian) [reply llvm/include/llvm/IR/IntrinsicsAMDGPU.td:2477]: My bad. Forgot to remove all of them.

File: llvm/include/llvm/IR/IntrinsicsAMDGPU.td [hunk 2]
```diff
@@ -2466,23 +2466,20 @@ def int_amdgcn_perm :
...
+                                        //                                   bit 1 = slc/sc1,
+                                        //                                   bit 4 = scc/nt on gfx90a+))
```
Comment (rampitec) [inline llvm/include/llvm/IR/IntrinsicsAMDGPU.td:2479]: Just sc0, sc1 and scc. It does not exist on gfx90a.

**Review by rampitec — APPROVED** [review]

File: clang/include/clang/Basic/BuiltinsAMDGPU.def [hunk 3]
```diff
@@ -240,6 +240,7 @@ TARGET_BUILTIN(__builtin_amdgcn_flat_atomic_fadd_v2bf16, "V2sV2s*0V2s", "t", "at
 TARGET_BUILTIN(__builtin_amdgcn_global_atomic_fadd_v2bf16, "V2sV2s*1V2s", "t", "atomic-global-pk-add-bf16-inst")
 TARGET_BUILTIN(__builtin_amdgcn_ds_atomic_fadd_v2bf16, "V2sV2s*3V2s", "t", "atomic-ds-pk-add-16-insts")
 TARGET_BUILTIN(__builtin_amdgcn_ds_atomic_fadd_v2f16, "V2hV2h*3V2h", "t", "atomic-ds-pk-add-16-insts")
+TARGET_BUILTIN(__builtin_amdgcn_global_load_lds, "vv*1v*3UiiUi", "t", "gfx940-insts")
```
Comment (arsenm) [inline clang/include/clang/Basic/BuiltinsAMDGPU.def:243]: clang should really be enforcing the valid immediate values for the size

**Pull request merged**